presto-parser/src/main/java/com/facebook/presto/sql/tree/FunctionCall.java

```diff
@@ -26,44 +26,60 @@
     private final Optional<Window> window;
     private final Optional<Expression> filter;
     private final boolean distinct;
+    private final boolean ignoreNulls;
     private final List<Expression> arguments;
 
     public FunctionCall(QualifiedName name, List<Expression> arguments)
     {
-        this(Optional.empty(), name, Optional.empty(),  Optional.empty(), false, arguments);
+        this(Optional.empty(), name, Optional.empty(), Optional.empty(), false, false, arguments);
     }
 
     public FunctionCall(NodeLocation location, QualifiedName name, List<Expression> arguments)
     {
-        this(Optional.of(location), name, Optional.empty(), Optional.empty(), false, arguments);
+        this(Optional.of(location), name, Optional.empty(), Optional.empty(), false, false, arguments);
     }
 
     public FunctionCall(QualifiedName name, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.empty(), name, Optional.empty(), Optional.empty(), distinct, arguments);
+        this(Optional.empty(), name, Optional.empty(), Optional.empty(), distinct, false, arguments);
     }
 
     public FunctionCall(NodeLocation location, QualifiedName name, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.of(location), name, Optional.empty(), Optional.empty(), distinct, arguments);
+        this(Optional.of(location), name, Optional.empty(), Optional.empty(), distinct, false, arguments);
     }
 
     public FunctionCall(QualifiedName name, Optional<Window> window, boolean distinct, List<Expression> arguments)
     {
-        this(Optional.empty(), name, window, Optional.empty(), distinct, arguments);
+        this(Optional.empty(), name, window, Optional.empty(), distinct, false, arguments);
+    }
+
+    public FunctionCall(QualifiedName name, Optional<Window> window, boolean distinct, boolean ignoreNulls, List<Expression> arguments)
```

Comment:
Where are all these new constructors used? This class already has too many, so I'd prefer to simply add as few as possible (or none).

I think we should add it as a parameter to all of the constructors that take a `Window` and update all the call sites to make sure it's passed through correctly. We have a lot of places that rewrite the AST, and it's easy for one of them to miss the parameter and silently change the behavior (thus producing wrong answers).